### ISSUE TITLE
Bug: RegExp type option don't throw validation error

### ISSUE THREAD

**Issue by snitin315** [eslint/eslint#20506]
### Environment

Playground

### What parser are you using?

Default (Espree)

### What did you do?

```js
<!-- Paste your code here -->

/*eslint no-restricted-imports: ["error", /fs/]*/

import fs from 'fs';
```

### What did you expect to happen?

Expected schema validation to throw error.

### What actually happened?

No error

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tcmVzdHJpY3RlZC1pbXBvcnRzOiBbXCJlcnJvclwiLCAvZnMvXSovXG5cbmltcG9ydCBmcyBmcm9tICdmcyc7Iiwib3B0aW9ucyI6eyJydWxlcyI6e30sImxhbmd1YWdlT3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19

### Participation

- [x] I am willing to submit a pull request for this issue.

### Additional comments

This might be a general issue with other rules as well.

**Comment by mdjermanovic** [comment]
I see the regex passes `type: "object"` schema validation because, technically, regex is an object. Any ideas on how this could be addressed at the schema validation level?

**Comment by DMartens** [comment]
Even with `type: "object"` the regular expression should fail the additional validation such as `properties` and `required`.
I think this may be a problem with the configuration normalization as in isolation it rejects the regular expression (see [stackblitz](https://stackblitz.com/edit/stackblitz-starters-a15mintg?file=index.js))

**Comment by mdjermanovic** [comment]
> Even with `type: "object"` the regular expression should fail the additional validation such as `properties` and `required`.
> I think this may be a problem with the configuration normalization as in isolation it rejects the regular expression (see [stackblitz](https://stackblitz.com/edit/stackblitz-starters-a15mintg?file=index.js))

It does fail `required` validation. For example, this fails [`required: ["name"]`](https://github.com/eslint/eslint/blob/04c21475b3004904948f02049f2888b401d82c78/lib/rules/no-restricted-imports.js#L75C5-L75C23):

https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tcmVzdHJpY3RlZC1pbXBvcnRzOiBbXCJlcnJvclwiLCB7IHBhdGhzOiBbL2ZzL10gfV0qL1xuXG5pbXBvcnQgZnMgZnJvbSAnZnMnOyIsIm9wdGlvbnMiOnsicnVsZXMiOnt9LCJsYW5ndWFnZU9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==

Example from the original post matches [`{ paths, patterns }`](https://github.com/eslint/eslint/blob/04c21475b3004904948f02049f2888b401d82c78/lib/rules/no-restricted-imports.js#L253-L266) definition.

**Comment by snitin315** [comment]
@mdjermanovic I think we can solve this via `ajv.addKeyword`, we need to add something like this:

```js
ajv.addKeyword("plainObjectOnly", {
		type: "object",
		validate(schema, data) {
			if (!schema) return true;
			return Object.prototype.toString.call(data) === "[object Object]";
		},
		errors: false,
	});
```

and pass `plainObjectOnly: true` to schema, so that we have following error:

<img width="636" height="108" alt="Image" src="https://github.com/user-attachments/assets/bf6d5905-bc99-4ab5-8861-2a550f537c00" />

**Comment by snitin315** [comment]
For reference https://github.com/eslint/eslint/pull/20545

**Comment by hayatexe** [comment]
Hi @snitin315! I've already developed a fix for this that addresses the RegExp literal validation issue. I'd love to contribute this fix. May I be assigned to this issue?

**Comment by snitin315** [comment]
Hi @hayatexe, thank you for your interest in contributing. However, I'm already working on it (https://github.com/eslint/eslint/pull/20545). Maybe you can help in reviewing my PR or share your proposed fix here.

**Comment by mdjermanovic** [comment]
There are some plugin rules that expect RegExp objects in options (e.g., https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/filename-case.md#ignore; this particular rule doesn't have `type: "object"` but there could be others that do have it specified in places where a RegExp object is expected). Although we don't officially support RegExp objects in options, I think disallowing them by throwing a validation error would be a breaking change.

**Comment by meganprice661-svg** [comment]
That makes sense. Removing RegExp objects could break things, especially
since some rules expect them. Maybe a clearer doc update would help instead
of throwing validation errors.

**Comment by nzakas** [comment]
I agree with @mdjermanovic -- whatever behavior we choose, it needs to align with the behavior of config files when a regular expression is provided as an option, which is the definitive and expected behavior for rule configuration. Currently I believe this does not fail validation but will fail serialization.

**Comment by meganprice661-svg** [comment]
If it fails serialization, should we still avoid validation errors then?

**Comment by snitin315** [comment]
To avoid breaking change, maybe we could add a new property to like `plainObjectOnly : boolean (default: false)` to `type: "object"` schema.

**Comment by mdjermanovic** [comment]
> To avoid breaking change, maybe we could add a new property to like `plainObjectOnly : boolean (default: false)` to `type: "object"` schema.

This solution would require changing the schemas in all rules for what seems like an edge case of misconfigurations, so I wouldn't be in favor of it. If we want to disallow passing RegExp objects where `type: "object"` is expected, I think it should just work by default (rules that do expect a RegExp object should remove `type: "object"`), but that kind of change would need to wait for ESLint v11.
